odata-protocol/11.4 Data Modification.md

```diff
@@ -657,17 +660,20 @@ the request.
 ### ##subsubsec Upsert an Entity
 
 An upsert occurs when the client sends an [update
-request](#UpdateanEntity) to a valid URL that identifies a single entity
-that does not yet exist. In this case the service MUST handle the
+request](#UpdateanEntity) to a valid URL for addressing a single entity [#OData-URL#AddressingEntities]
+and the requested update newly creates the addressed entity.
+In this case the service MUST handle the
 request as a [create entity request](#CreateanEntity) or fail the
 request altogether.
 
-Upserts to single-valued navigation properties are possible for
-
-- containment navigation properties,
-- non-containment navigation properties with a navigation property binding, or
-- payloads including a context URL specifying the entity set or
-  contained collection of entities in which the new entity is to be created.
+Upserts are possible if the canonical collection or canonical singleton
+(as defined in [section ##ContextURL]) can be determined for the newly-created entity.
+Among other possibilities, this is the case if
+1. the URL contains the name of an entity set or singleton
```

Comment:
More precision needed? "URL contains a segment consisting of the name of an entity set or singleton"